enginetest/script_queries.go

```diff
@@ -755,4 +755,157 @@ var ScriptTests = []ScriptTest{
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check1` CHECK ((`pk` = 5)),\n" +
+							"  CONSTRAINT `check11` CHECK ((`pk` < 6))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable2",
+				Expected: []sql.Row{
+					{
+						"mytable2",
+						"CREATE TABLE `mytable2` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check2` CHECK ((`v` < 5)),\n" +
+							"  CONSTRAINT `check12` CHECK (((`pk` + `v`) = 6))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable3",
+				Expected: []sql.Row{
+					{
+						"mytable3",
+						"CREATE TABLE `mytable3` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check3` CHECK (((`pk` > 2) AND (`v` < 5))),\n" +
+							"  CONSTRAINT `check13` CHECK ((`pk` BETWEEN 2 AND 100))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable4",
+				Expected: []sql.Row{
+					{
+						"mytable4",
+						"CREATE TABLE `mytable4` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check4` CHECK ((((`pk` > 2) AND (`v` < 5)) AND (`pk` < 9)))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable5",
+				Expected: []sql.Row{
+					{
+						"mytable5",
+						"CREATE TABLE `mytable5` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check5` CHECK (((`pk` > 2) OR ((`v` < 5) AND (`pk` < 9))))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable6",
+				Expected: []sql.Row{
+					{
+						"mytable6",
+						"CREATE TABLE `mytable6` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check6` CHECK (NOT(`pk`))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable7",
+				Expected: []sql.Row{
+					{
+						"mytable7",
+						"CREATE TABLE `mytable7` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check7` CHECK (NOT((`pk` = `v`)))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			{
+				Query: "SHOW CREATE TABLE mytable8",
+				Expected: []sql.Row{
+					{
+						"mytable8",
+						"CREATE TABLE `mytable8` (\n  `pk` int NOT NULL,\n" +
+							"  `v` int,\n" +
+							"  PRIMARY KEY (`pk`),\n" +
+							"  CONSTRAINT `check8` CHECK ((((`pk` > 2) OR (`v` < 5)) OR (`pk` < 10)))\n" +
+							") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
+					},
+				},
+			},
+			// TODO: Expression tree is Incorrect
```

Comment:
?

Reply:
Bad comment on my part. I should've been explained that I thought this caused a bug on the expression tree. I just double checked and it's no longer a problem